tyrian/shared/src/main/scala/tyrian/Attr.scala

```diff
@@ -86,5 +86,29 @@ object Property:
   * @param msg
   *   Message to produce when the event is triggered
   */
-final case class Event[E <: Tyrian.Event, M](name: String, msg: E => M) extends Attr[M]:
-  def map[N](f: M => N): Attr[N] = Event(name, msg andThen f)
+final case class Event[E <: Tyrian.Event, M](
+    name: String,
+    msg: E => M,
+    preventDefault: Boolean,
+    stopPropagation: Boolean,
+    stopImmediatePropagation: Boolean
+) extends Attr[M]:
+  def map[N](f: M => N): Attr[N]                  = Event(name, msg andThen f)
+  def preventDefaultToggle: Event[E, M]           = this.copy(preventDefault = !preventDefault)
+  def preventDefaultOn: Event[E, M]               = this.copy(preventDefault = true)
+  def preventDefaultOff: Event[E, M]              = this.copy(preventDefault = false)
+  def stopPropagationToggle: Event[E, M]          = this.copy(stopPropagation = !stopPropagation)
+  def stopPropagationOn: Event[E, M]              = this.copy(stopPropagation = true)
```

Comment:
Likewise for these ... just brings it inline with Indigo really, but happy to go either way :)

Reply:
Is Indigo's approach the best - don't know, but bringing it inline isn't a bad shout. That would mean this sort of thing:

```
  def withTransparentBackground(enabled: Boolean): GameConfig =
    this.copy(transparentBackground = enabled)
  def useTransparentBackground: GameConfig =
    withTransparentBackground(true)
  def noTransparentBackground: GameConfig =
    withTransparentBackground(false)
```

`with`/`use`/`no`.

I can't decide whether one is better than the other, but I'll bow to consistency.